packages/sanity/src/core/store/key-value/backends/server.ts

```diff
@@ -0,0 +1,117 @@
+import {type SanityClient} from '@sanity/client'
+import DataLoader from 'dataloader'
+import {catchError, from, map, of} from 'rxjs'
+
+import {DEFAULT_STUDIO_CLIENT_OPTIONS} from '../../../studioClient'
+import {type KeyValueStoreValue} from '../types'
+import {type Backend, type KeyValuePair} from './types'
+
+/** @internal */
+export interface ServerBackendOptions {
+  client: SanityClient
+}
+
+/**
+ * One of serveral possible backends for KeyValueStore. This backend uses the
+ * Sanity client to store and retrieve key-value pairs from the /users/me/keyvalue endpoint.
+ * @internal
+ */
+export function serverBackend({client: _client}: ServerBackendOptions): Backend {
+  const client = _client.withConfig({...DEFAULT_STUDIO_CLIENT_OPTIONS, apiVersion: 'vX'})
+
+  const keyValueLoader = new DataLoader<string, KeyValueStoreValue | null>(async (keys) => {
+    const value = await client
+      .request<KeyValuePair[]>({
+        uri: `/users/me/keyvalue/${keys.join(',')}`,
+        withCredentials: true,
+      })
+      .catch((error) => {
+        if (error.response?.statusCode === 404) {
+          return Array(keys.length).fill(null)
+        }
+        throw error
+      })
+    const keyValuePairs = value.reduce(
+      (acc, next) => {
+        if (next?.key) {
+          acc[next.key] = next.value
+        }
+        return acc
+      },
+      {} as Record<string, KeyValueStoreValue | null>,
+    )
+
+    const result = keys.map((key) => keyValuePairs[key] || null)
```

Comment:
I don't believe the backend is guaranteed to return key/value pairs in the order that they're requested, and this same method is used in userStore.

Reply:
I can do that

Reply:
On revisiting this, the studio is quite fragile here. Even handling defaults in hooks and where they're being consumed does not help -- the Dataloader causes quite a dramatic error and renders the studio unusable.

For better peace of mind, I recommend that we keep this as an extra layer of handling any problems or upstream changes gracefully.